crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -318,6 +317,9 @@ impl<'a> WasiSnapshotPreview1 for WasiCtx {
             }
 
             buf = buf.add(dirent_copy_len)?;
+            bufused = bufused
+                .checked_add(dirent_copy_len)
+                .ok_or(Error::Overflow)?;
```

Comment:
the `buf.add(dirent_copy_len)?` performs this overflow check, so bufused doesnt have to have a checked add